### PR TITLE
ODIM-5021 Implement wild card for metric properties

### PR DESCRIPTION
**Pull request by irappa-pattar**

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]

**Review by skhan-anjum — APPROVED** [review]

File: svc-aggregation/system/common.go [hunk 1]
```diff
@@ -1168,11 +1181,13 @@ func updateResourceDataWithUUID(resourceData, uuid string) string {
 	//replacing the uuid while saving the data
 	//to replace the id of system
 	var updatedResourceData = strings.Replace(resourceData, "/redfish/v1/Systems/", "/redfish/v1/Systems/"+uuid+":", -1)
-	updatedResourceData = strings.Replace(updatedResourceData, "/redfish/v1/systems/", "/redfish/v1/systems/"+uuid+":", -1)
+	updatedResourceData = strings.Replace(updatedResourceData, "/redfish/v1/systems/", "/redfish/v1/Systems/"+uuid+":", -1)
 	// to replace the id in managers
 	updatedResourceData = strings.Replace(updatedResourceData, "/redfish/v1/Managers/", "/redfish/v1/Managers/"+uuid+":", -1)
 	// to replace id in chassis
-	return strings.Replace(updatedResourceData, "/redfish/v1/Chassis/", "/redfish/v1/Chassis/"+uuid+":", -1)
+	updatedResourceData = strings.Replace(updatedResourceData, "/redfish/v1/Chassis/", "/redfish/v1/Chassis/"+uuid+":", -1)
```
Comment (amar-shalgar) [inline svc-aggregation/system/common.go:1188]: Please remove the duplicate code for replacing chassis

File: svc-aggregation/system/common.go [hunk 2]
```diff
@@ -1436,3 +1459,138 @@ func getTeleInfo(taskID string, progress, alottedWork int32, req getResourceRequ
 	req.UpdateTask(task)
 	return progress
 }
+
+func createWildCard(resourceData, resourceName, oid string) (string, error) {
```
Comment (amar-shalgar) [inline svc-aggregation/system/common.go:1465]: Please add comment for the function

File: svc-aggregation/system/common.go [hunk 3]
```diff
@@ -1436,3 +1459,138 @@ func getTeleInfo(taskID string, progress, alottedWork int32, req getResourceRequ
...
+	err := json.Unmarshal([]byte(resourceData), &resourceDataMap)
+	if err != nil {
+		log.Error("Failed to unmarshal the resource data, got: " + err.Error())
+		return "", err
+	}
+	data, _ := agmodel.GetResource(resourceName, oid)
+	return formWildCard(data, resourceDataMap)
+}
+
+func formWildCard(dbData string, resourceDataMap map[string]interface{}) (string, error) {
```
Comment (amar-shalgar) [inline svc-aggregation/system/common.go:1479]: Add comment to the func

**Review by amar-shalgar — CHANGES_REQUESTED** [review]
Please address comments

File: svc-aggregation/system/common.go (hunk 2 above)
Comment (irappa-pattar) [reply svc-aggregation/system/common.go:1465]: addressed

File: svc-aggregation/system/common.go (hunk 3 above)
Comment (irappa-pattar) [reply svc-aggregation/system/common.go:1479]: addressed

File: svc-aggregation/system/common.go (hunk 1 above)
Comment (irappa-pattar) [reply svc-aggregation/system/common.go:1188]: its not duplicate, one line is for replacing **chassis** and another **Chassis**

**Review by amar-shalgar — APPROVED** [review]
LGTM

**Review by bharath-b-hpe — APPROVED** [review]

**Pull request merged**